backend/open_api_spec.yml

```diff
@@ -0,0 +1,829 @@
+        content:
+          application/json:
+            schema:
+              type: object
+              additionalProperties: false
+              properties:
+                curator_tag:
+                  type: string
+                id:
+                  $ref: "#/components/schemas/dataset_uuid"
+                link:
+                  $ref: "#/components/schemas/upload_link"
+      responses:
+        "204":
+          $ref: "#/components/responses/204"
+        "401":
+          $ref: "#/components/responses/401"
+        "404":
+          $ref: "#/components/responses/404"
+
+  /curation/collections/{collection_id}/datasets/upload-s3:
+    put:
+      description: >
+        Retrieve AWS credentials for uploading Dataset source files to S3 to create
+        or update Datasets for the specified Collection.
+      security:
+        - cxguserCookie: []
+      parameters:
+        - $ref: "#/components/parameters/path_collection_uuid"
+      responses:
+        "200":
+          description: OK
+          content:
+            application/json:
+              schema:
+                type: object
+                additionalProperties: false
+                properties:
+                  credentials:
+                    type: object
+                    properties:
+                      access_key:
+                        type: string
+                        required: true
+                      secret_access_key:
+                        type: string
+                        required: true
+                      access_token:
+                        type: string
+                        required: true
+        "401":
+          $ref: "#/components/responses/401"
+        "404":
+          $ref: "#/components/responses/404"
+
+  /curation/collections/{collection_id}/datasets/status:
+    get:
+      description: Get the processing status for a Dataset.
+      security:
+        - cxguserCookie: []
+      parameters:
+        - $ref: "#/components/parameters/path_collection_uuid"
+        - $ref: "#/components/parameters/optional_query_curator_tag"
+        - $ref: "#components/parameters/optional_query_dataset_uuid"
+      responses:
+        "200":
+          description: OK
+          content:
+            application/json:
+              schema:
+                type:
+                  object:
+                    properties:
+                      status:
+                        type: string
+                        enum: [FAILURE, PENDING, SUCCESS]
+                        required: true
+
+components:
+  schemas:
+    user_uuid:
+      description: A unique identifier of a logged in User of Corpora.
+      type: string
+    userinfo:
+      type: object
+      properties:
+        id:
+          $ref: "#/components/schemas/user_uuid"
+        name:
+          description: The user's name
+          type: string
+        email:
+          description: The user's email
+          type: string
+    collection_metadata:
```

Comment:
Remove published_at and revised_at. These are not fields the user has permission to modify. Make a separate schema for the collection metadata response if it will include the additional fields. Be careful when reusing schemas. The schema conveys meaning in the backend and could result in us giving the users more privilege than we intend.

Reply:
Great point. The schemas need to be different for request and response.